src/sparse/CSRMatrix.cpp

```diff
@@ -1160,6 +1160,32 @@ namespace strumpack {
     return CSRGraph<integer_t>(std::move(gptr), std::move(gind));
   }
 
+  template<typename scalar_t, typename integer_t, typename cast_t> 
+  CSRMatrix<cast_t,integer_t> 
+  cast_matrix(const CSRMatrix<scalar_t,integer_t>& mat) {
+    const integer_t nnz = mat.nnz();
+    const integer_t size = mat.size();
+    CSRMatrix<cast_t,integer_t> new_mat(size, nnz);
```

Comment:
You can use the other constructor here:
```
std::vector<cast_t> new_val(mat.val(), mat.val()+nnz); // this will take care of the cast
CSRMatrix<cast_t,integer_t> new_mat(size, mat.ptr(), mat.ind(), new_val, mat.symm_sparse());
```
That will also make copies of the ptr and ind arrays in mat.

Reply:
Oh, thanks! This is a much cleaner solution :) switched to the other constructor.